Test Failures dom-slot set content

Environment
- macos: 10.13.4
- nodejs 10.4.0 and npm 6.1.0 installed from nodejs.org

Running `./tools/sigh` results in a failure in `runtime/test/dom-slot-tests.js`

```
dom-slot set content:
     AssertionError: expected { foo: 'bar' } to deeply equal null
```

Adding a bunch of tracing resulted in identifying an await call that returns before modifications are made.

```
   if (Object.keys(content).indexOf('model') >= 0) {
      if (content.model) {
        this._model = Object.assign(content.model, await this.populateHandleDescriptions());
      } else {
        this._model = undefined;
      }
    }
```

`populateHandleDescriptions` is redefined in the test to an empty async function.

```
    slot.populateHandleDescriptions = async () => {};
```

Removing the async causes this test to pass.  I have not identified the underlying reason that this happens.

--- a/runtime/test/dom-slot-tests.js
+++ b/runtime/test/dom-slot-tests.js
@@ -55,7 +55,7 @@ describe('dom-slot', function() {
       ++doRenderCount;
       _doRenderImpl.call(slot);
     };
-    slot.populateHandleDescriptions = async () => {};
+    slot.populateHandleDescriptions = () => {};
     assert.isNull(slot._model);
 
     // model and context are null; set content to null - nothing happens.

